### PR TITLE
refactor: scheduling Hasura metadata relationships

### PR DESCRIPTION
**Pull request by camargo**
## Description

Updates Hasura metadata for client querying.

- Add relationship from a plan to associated scheduling specifications
- Add relationship from specification goals to associated goal and spec

### CODE REVIEW

File: deployment/hasura/metadata/databases/AerieScheduler/tables/public_scheduling_specification_goals.yaml [hunk 1]
```diff
@@ -1,3 +1,10 @@
 table:
   name: scheduling_specification_goals
   schema: public
+object_relationships:
```
Comment (mattdailis) [inline deployment/hasura/metadata/databases/AerieScheduler/tables/public_scheduling_specification_goals.yaml:4]: I'm curious that the arrow is going from `scheduling_specification_goals` -> `specification` and not the other way around. What query is this intended to make easier?

File: deployment/hasura/metadata/databases/AerieMerlin/tables/public_plan.yaml [hunk 2]
```diff
@@ -27,3 +27,14 @@ array_relationships:
       table:
         name: simulation
         schema: public
+remote_relationships:
```
Comment (mattdailis) [inline deployment/hasura/metadata/databases/AerieMerlin/tables/public_plan.yaml:30]: Remote relationships is such a cool idea

**Review by mattdailis — APPROVED** [review]

File: deployment/hasura/metadata/databases/AerieScheduler/tables/public_scheduling_specification_goals.yaml (hunk 1 above)
Comment (camargo) [reply deployment/hasura/metadata/databases/AerieScheduler/tables/public_scheduling_specification_goals.yaml:4]: The UI displays the "goal list" as specification goals so we can include the priority. This gives clients a much easier way to get the full specification goal definition that includes the associated goal, or the associated specification, or any combination of both.

Comment (camargo) [reply deployment/hasura/metadata/databases/AerieScheduler/tables/public_scheduling_specification_goals.yaml:4]: That said, we could easily make a relationship the other way if the need arises.

**Review by pcrosemurgy — APPROVED** [review]
Thanks Chris!

**Pull request merged**